src/containers/Accounts/AMM/AMMAccounts/index.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { useParams, useRouteMatch } from 'react-router'
+import { useTranslation } from 'react-i18next'
+import { connect } from 'react-redux'
+import AMMAccountHeader from 'containers/Accounts/AMM/AMMAccounts/AMMAccountHeader/AMMAccountHeader'
+import { AccountTransactionTable } from 'containers/Accounts/AccountTransactionTable/index'
+import NoMatch from 'containers/NoMatch'
+import 'containers/Accounts/styles.scss'
+import {
+  analytics,
+  ANALYTIC_TYPES,
+  formatAsset,
+  formatBalance,
+} from 'containers/shared/utils'
+import { Tabs } from 'containers/shared/components/Tabs'
+import { AccountAssetTab } from 'containers/Accounts/AccountAssetTab/AccountAssetTab'
+import { getAccountTransactions, getAMMInfo } from 'rippled/lib/rippled'
+import SocketContext from '../../../shared/SocketContext'
+import { ERROR_MESSAGES } from '../../Errors'
+import { getTokenPairData } from '../../../../apis/OnTheDex'
+
+export interface AmmDataType {
+  balance: { currency: string; amount: number; issuer: string }
+  balance2: { currency: string; amount: number; issuer: string }
+  lpBalance: number
+  ammId: string
+  tradingFee: number
+  accountId: string
+  language: string
+  tvl?: number
+}
+
+const getErrorMessage = (error: string) =>
+  ERROR_MESSAGES[error] || ERROR_MESSAGES.default
+
+const USDGatehub = {
+  currency: 'USD',
+  issuer: 'rhub8VRN55s94qWKDv6jmDy1pUykJzF3wq',
+}
+
+const AMMAccounts = (props: any) => {
+  const { id: accountId, tab = 'transactions' } = useParams<{
+    id: string
+    tab: string
+  }>()
+  const { path = '/' } = useRouteMatch()
+  const { t } = useTranslation()
+  const mainPath = `${path.split('/:')[0]}/${accountId}`
+  const [currencySelected] = useState('XRP')
+  const rippledSocket = useContext(SocketContext)
+  const [data, setData] = useState<AmmDataType>()
+  const [error, setError] = useState<any>()
+  const [tvl, setTVL] = useState<any>()
+  const { language } = props
+
+  useEffect(() => {
+    let asset1: { currency: string; issuer?: string }
+    let asset2: { currency: string; issuer?: string }
+    let volumeAsset1: number
+    let volumeAsset2: number
+    let assetXrpValue: number
+    let asset2XrpValue: number
+    let ammData: any
+    let balance: any
+    let balance2: any
```

Comment:
Why are these all `let` before? Seems unnecessary, they're never changed and only created/used once

Reply:
I needed a way to pass the data down the promise chain and I didn't know how else to do it. We need to access the values from one network call/promise to another.